BXMSDOC-5513-master-kogito: Incorporate Edoardo's changes to DRL, XLS, decision engine for Kogito (#2324)
kiegroup/kie-docs: doc-content/kogito-docs/src/main/asciidoc/drl/con_drl-rule-units.adoc

= Rule units in DRL

> A DRL rule unit is a module for rules and a unit of execution. A rule unit collects a set of rules with the declaration of the type of facts that the rules act on. A rule unit also serves as a unique namespace for each group of rules. A single rule base can contain multiple rule units. You typically store all the rules for a unit in the same file as the unit declaration so that the unit is self-contained.

Reviewer: I know that you are trying to go with consistency for the use case, and sorry I introduced a different example here, but the result is a bit confusing. The flow of my original piece:

- first declare data types (alerts and temperatures; in your case Perrson
- then declare the "schema" of the data that will be used, by declaring the data sources
- then declare the rules

unit MortgageRules;
----

> To define a rule unit, you add a DRL file with data type declarations that serves as a rule unit class, and then implement the declared data and data sources using the `RuleUnitData` interface in your DRL rule file, as shown in the following examples:

Reviewer: ```suggestion
To define a rule unit, you you may use a data type declaration implementing `RuleUnitData` interface.
The data type will declare the data sources as shown in the following examples:
```

> .Example DRL rule unit class to define declarations

Reviewer: this is not a rule unit. it does not implement RuleUnitData. This is just a type declaration for Person

[source]
----
package org.mortgages;

declare Person
    name : String
    dateOfBirth : Date
    address : Address
end
----

> .Example DRL rule file to implement rule unit and declarations

Reviewer: this DRL declares the rule unit "type" or schema ("declare MortgageRules extends RuleUnitData"), or in other words, it declares the list of data sources using the "declare" syntax;
then there is a rule that uses the included data source declarations.
